### PR TITLE
DataGrid: update calculateFilterExpression code snippet (T1158674)

### PR DESCRIPTION
**Pull request by sergepilipchuk**

### CODE REVIEW

File: api-reference/_hidden/GridBaseColumn/calculateFilterExpression.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ The default *"between"* implementation is inclusive (filter results include the
         $("#{widgetName}Container").dx{WidgetName}({
             // ...
             columns: [{
-                calculateFilterExpression: function (filterValue, selectedFilterOperation) {
+                calculateFilterExpression: function (Column, filterValue, selectedFilterOperation, target) {
```
Comment (artem-kurchenko) [inline api-reference/_hidden/GridBaseColumn/calculateFilterExpression.md]: no "Column" parameter in jQuery

Comment (sergepilipchuk) [reply api-reference/_hidden/GridBaseColumn/calculateFilterExpression.md]: Ok

File: api-reference/_hidden/GridBaseColumn/calculateFilterExpression.md [hunk 2]
```diff
@@ -75,19 +77,20 @@ The default *"between"* implementation is inclusive (filter results include the
     import { Dx{WidgetName}Module } from "devextreme-angular";
     // ...
     export class AppComponent {
-        calculateFilterExpression (filterValue, selectedFilterOperation) {
-            const column = this as any;
+        calculateFilterExpression (this: Column, filterValue, selectedFilterOperation, target) {
```
Comment (artem-kurchenko) [inline api-reference/_hidden/GridBaseColumn/calculateFilterExpression.md:81]: The Column object import is missing, see this demo as an example:
https://js.devexpress.com/Demos/WidgetsGallery/Demo/DataGrid/CascadingLookups/Angular/Light/

File: api-reference/_hidden/GridBaseColumn/calculateFilterExpression.md [hunk 3]
```diff
@@ -159,7 +163,7 @@ The default *"between"* implementation is inclusive (filter results include the
         Column
     } from 'devextreme-react/{widget-name}';
 
-    function calculateFilterExpression (filterValue, selectedFilterOperation) {
+    function calculateFilterExpression (this: Column, filterValue, selectedFilterOperation, target) {
```
Comment (artem-kurchenko) [inline api-reference/_hidden/GridBaseColumn/calculateFilterExpression.md]: it's not a TS code, you can remove this: Column

File: api-reference/_hidden/GridBaseColumn/calculateFilterExpression.md [hunk 4]
```diff
@@ -130,19 +133,20 @@ The default *"between"* implementation is inclusive (filter results include the
         },
         data() {
             return {
-                calculateFilterExpression (filterValue, selectedFilterOperation) {
-                    const column = this;
+                calculateFilterExpression (this: Column, filterValue, selectedFilterOperation, target) {
```
Comment (artem-kurchenko) [inline api-reference/_hidden/GridBaseColumn/calculateFilterExpression.md]: it's not a TS code, you can remove this: Column

**Review by artem-kurchenko — APPROVED** [review]

**Pull request merged**